packages/logs/src/domain/trackReports.ts

```diff
@@ -0,0 +1,105 @@
+import { DOM_EVENT, addEventListener, monitor } from '@datadog/browser-core'
+
+export interface BrowserWindow {
+  ReportingObserver?: ReportingObserver
+}
+
+export interface ReportingObserver {
+  disconnect(): void
+  observe(): void
+  takeRecords(): Report[]
+}
+
+export interface ReportingObserverOption {
+  types: ReportType[]
+  buffered: boolean
+}
+
+interface ReportingObserverCallback {
+  (reports: Report[], observer: ReportingObserver): void
+}
+
+declare const ReportingObserver: {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏maybe `declare class ReportingObserver {` would be more idiomatic?

```ts
declare class ReportingObserver extends PerformanceObserver {
  static readonly supportedEntryTypes: readonly string[]
  constructor(callback: ReportingObserverCallback, option: ReportingObserverOption)
  disconnect(): void
  observe(): void
  takeRecords(): Report[]
}
```